components/api/src/main/java/com/hotels/styx/api/HttpRequest.java

```diff
@@ -335,7 +335,7 @@ private FormData toFormData(ByteBuf byteBuf) {
      */
     public Optional<HttpCookie> cookie(String name) {
         return cookies().stream()
-                .filter(cookie -> name.equalsIgnoreCase(cookie.name()))
+                .filter(cookie -> name.equals(cookie.name()))
                 .findFirst();
     }
 
```

Comment:
I would add few tests to demonstrate that the cookie names indeed are case insensitive.